eng/scripts/Language-Settings.ps1

```diff
@@ -1,82 +1,127 @@
+{
+    $workFolder = $pkg.Directory
+    $releaseNotes = ""
 
-    $changeLogLoc = @(Get-ChildItem -Path $workFolder -Recurse -Include "CHANGELOG.md")[0]
-    if ($changeLogLoc)
+    $packageProperties = (Get-AllPackageInfoFromRepo -pkgDirectory $workFolder)[0]
+
+    if ($packageProperties.ChangeLogPath -and $packageProperties.Version)
     {
-        $releaseNotes = Get-ChangeLogEntryAsString -ChangeLogLocation $changeLogLoc -VersionString v$pkgVersion
+      $releaseNotes = Get-ChangeLogEntryAsString -ChangeLogLocation $packageProperties.ChangeLogPath `
+        -VersionString $packageProperties.Version
     }
 
     $resultObj = New-Object PSObject -Property @{
-        PackageId      = $pkgId
-        PackageVersion = $pkgVersion
-        ReleaseTag     = "$pkgId/v$pkgVersion"
-        Deployable     = $true
-        ReleaseNotes   = $releaseNotes
+      PackageId      = $packageProperties.Name
+      PackageVersion = $packageProperties.Version
+      ReleaseTag     = "$($packageProperties.Name)/v$($packageProperties.Version)"
+      Deployable     = $true
+      ReleaseNotes   = $releaseNotes
     }
 
     return $resultObj
 }
 
-# get version from specific files (*constants.go, *version.go)
-function Get-Version ($pkgPath)
+function Get-AllPackageInfoFromRepo ($serviceDirectory, $pkgDirectory)
 {
-    # find any file with suffix
-    $versionFiles = @()
-    $version_file_suffixs = "*constants.go", "*version.go"
-    foreach ($versionFileSuffix in $version_file_suffixs)
-    {
-        Get-ChildItem -Recurse -Path $pkgPath -Filter $versionFileSuffix | ForEach-Object {
-            Write-Host "Adding $_ to list of version files"
-            $versionFiles += $_
-        }
-    }
-    
-    # for each version file, use regex to search go version num
-    $go_version_regex = ".+\s*=\s*`".*v?(?<version>$([AzureEngSemanticVersion]::SEMVER_REGEX))`""
-    foreach ($versionFile in $versionFiles)
-    {
-        try
-        {
-            $content = Get-Content $versionFile -Raw
-            # finding where the version number are
-            if ($content -match $go_version_regex)
-            {
-                return $matches["version"]
-            }
-        }
-        catch
-        {
-            Write-Error "Error parsing version."
-            Write-Error $_
-        }
+  $allPackageProps = @()
+  $searchPath = Join-Path $RepoRoot "sdk"
+  if ($serviceDirectory) {
+    $searchPath = Join-Path $searchPath $serviceDirectory
+  }
+
+  if ($pkgDirectory) {
+    $searchPath = Join-Path $searchPath $pkgDirectory
+  }
+
+  $pkgFiles = Get-ChildItem -Path $searchPath -Include "go.mod" -Recurse
+
+  foreach ($pkgFile in $pkgFiles)
+  {
+    $modPropertes = Get-GoModuleProperties $pkgFile.DirectoryName
+
+    if ($modPropertes) {
+      $allPackageProps += $modPropertes
     }
+  }
+  return $allPackageProps
+}
+
+function SetPackageVersion ($PackageName, $Version, $ReleaseDate, $PackageProperties)
```

Comment:
Will there never be a need for version increment? Looks like you removed that function.

Reply:
Version increment will likely be something we want at some point but for now I wanted to keep this simple and focused and we don't currently even know what incrementing will look like for go yet.